### PR TITLE
multi: add shadow widget to debug and about page

### LINKED ISSUE #818: Add shadow widget to debug page and about page
Add shadow widget to the about page as hovering on the list item, makes the page layout fade. see screen recording.
something similar has been implemented on the transactions page.

ezgif com-gif-maker(10)

Add shadow to the about page license button. 
image

### PR DESCRIPTION
**Pull request by AdimekweEbuka**
Resolves #818 
godcr (testnet) 2022-03-26 08-31-59

### CODE REVIEW

**Review by devchoplife — CHANGES_REQUESTED** [review]
The page layout still fades on Debug and About page.
Debug page:
Screenshot from 2022-03-26 09-13-34

About page:
Screenshot from 2022-03-26 09-13-45

Looking at the More page, I believe this is more suitable as the shadow completely wraps the row and makes it distinct when hovered on.
More Page:
Screenshot from 2022-03-26 09-12-04

**Comment by AdimekweEbuka** [comment]
Regarding @devchoplife  review, I was able to recreate the debug page following the more page and I was able to get the shadow effect exactly as requested. See below: 
godcr (testnet) 2022-03-28 21-30-25


>However this is not the design that was on the original debug page, as on the original page there was no space between rows, inorder to get that design I removed the margins and after doing so the shadow effect got reverted to this. Please see below: 
godcr (testnet) 2022-03-28 21-33-53

>I have noticed that in pages where the rows are smooched together like that the shadow effect is faded but when seperated, its more prominent. Please see the transaction page, about page and recent transactions on the overview page: 

godcr (testnet) 2022-03-28 21-31-37 (1)


>I'm having trouble figuring out the reason for this discrepancy. 
>@SirMorrison Please can we get your take on this.

File: ui/page/debug_page.go [hunk 1]
```diff
@@ -13,47 +13,51 @@ import (
...
+	action    func()
+	clickable *decredmaterial.Clickable
 }
 
 type DebugPage struct {
 	*load.Load
+	container  layout.Flex
+	shadowBox  *decredmaterial.Shadow
 	debugItems []debugItem
-	list       *decredmaterial.ClickableList
```
Comment (Sirmorrison) [inline ui/page/debug_page.go]: Undo this. The clickable list has `IsShadowEnabled` Param. Use it instead see example use case here.
.https://github.com/planetdecred/godcr/blob/928443619efe397e68133e2692ca5ee1675bb315/ui/page/transaction/transactions_page.go#L51.

image

File: ui/page/about_page.go [hunk 2]
```diff
@@ -129,6 +131,25 @@ func (pg *AboutPage) layoutRows(gtx layout.Context) layout.Dimensions {
...
+									})
+								})
+							}),
+						)
+					})
+				})
+			}
 			return pg.licenseRow.Layout(gtx, func(gtx C) D {
 				return layout.Flex{}.Layout(gtx,
 					layout.Rigid(func(gtx C) D {
```
Comment (Sirmorrison) [inline ui/page/about_page.go]: You are repeating codes. At the very best avoid repeating codes. instead use function and call them where needed.

```suggestion
			licenseRow := func(gtx C) D {
				return pg.licenseRow.Layout(gtx, func(gtx C) D {
					return layout.Flex{}.Layout(gtx,
						layout.Rigid(func(gtx C) D {
							return in.Layout(gtx, pg.license.Layout)
						}),
						layout.Flexed(1, func(gtx C) D {
							return layout.E.Layout(gtx, func(gtx C) D {
								return in.Layout(gtx, func(gtx C) D {
									pg.chevronRightIcon.Color = pg.Theme.Color.Gray1
									return pg.chevronRightIcon.Layout(gtx, values.MarginPadding20)
								})
							})
						}),
					)
				})
			}

			if pg.licenseRow.IsHovered() {
				return pg.shadowBox.Layout(gtx,licenseRow)
			}
			return  licenseRow(gtx)
		},
	}

	return pg.container.Layout(gtx, len(w), func(gtx C, i int) D {
		return layout.Inset{Bottom: values.MarginPadding3}.Layout(gtx, func(gtx C) D {
```
Take note of the insert Bottom  added.

**Review by Sirmorrison — CHANGES_REQUESTED** [review]

File: ui/page/about_page.go [hunk 3]
```diff
@@ -164,6 +157,24 @@ func (pg *AboutPage) layoutRows(gtx layout.Context) layout.Dimensions {
...
+				return layout.E.Layout(gtx, func(gtx C) D {
+					return in.Layout(gtx, func(gtx C) D {
+						pg.chevronRightIcon.Color = pg.Theme.Color.Gray1
+						return pg.chevronRightIcon.Layout(gtx, values.MarginPadding20)
+					})
+				})
+			}),
+		)
+	})
+}
```
Comment (Sirmorrison) [inline ui/page/about_page.go]: make this an inline func() by moving it to the section it was used. 

It was only ever used within that section. I don't see the need for this. except there is a reason though kindly share

File: ui/page/about_page.go [hunk 4]
```diff
@@ -129,21 +131,12 @@ func (pg *AboutPage) layoutRows(gtx layout.Context) layout.Dimensions {
...
-						})
-					}),
-				)
-			})
+			if pg.licenseRow.IsHovered() {
+				return pg.shadowBox.Layout(gtx, func(gtx C) D {
+					return pg.licenseRowLayout(gtx, in)
+				})
+			}
+			return pg.licenseRowLayout(gtx, in)
```
Comment (Sirmorrison) [inline ui/page/about_page.go]:
```suggestion
licenseRow := func(gtx C) D {
				return pg.licenseRow.Layout(gtx, func(gtx C) D {
					return layout.Flex{}.Layout(gtx,
						layout.Rigid(func(gtx C) D {
							return in.Layout(gtx, pg.license.Layout)
						}),
						layout.Flexed(1, func(gtx C) D {
							return layout.E.Layout(gtx, func(gtx C) D {
								return in.Layout(gtx, func(gtx C) D {
									pg.chevronRightIcon.Color = pg.Theme.Color.Gray1
									return pg.chevronRightIcon.Layout(gtx, values.MarginPadding20)
								})
							})
						}),
					)
				})
			}
			if pg.licenseRow.IsHovered() {
				return pg.shadowBox.Layout(gtx,licenseRow)
			}
			return  licenseRow(gtx)
		},
	}
	return pg.container.Layout(gtx, len(w), func(gtx C, i int) D {
		return layout.Inset{Bottom: values.MarginPadding3}.Layout(gtx, func(gtx C) D {
```

Comment (Sirmorrison) [reply ui/page/about_page.go]: very important... note `return layout.Inset{Bottom: values.MarginPadding3}`

**Review by Sirmorrison — CHANGES_REQUESTED** [review]
This looks ok from the test. Just a little change and we are good to go.

**Comment by AdimekweEbuka** [comment]
Please note adding the bottom margin padding causes the license row to look like this when its hovered. 
godcr (testnet) 2022-03-30 21-29-16

**Review by Sirmorrison — APPROVED** [review]

**Pull request merged**